src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -681,6 +696,17 @@ private object GetGraphType(Type clrType, bool input, List<(Type clr, Type graph
                 }
             }
 
+            // then an available IGraphTypeMapping instance
+            if (_graphTypeMappings != null)
```

Comment:
I have a feeling that we may merge new API with `TypeMappings`. I mean delete `TypeMappings` and stop passing it down the `SchemaTypes` here to **unify** API. So `Schema.RegisterTypeMapping` registers another `IGraphTypeMapping` and then code works only with `IGraphTypeMapping` instances here. Drawback - someone may want to know what types are registered via `public Schema.TypeMappings` property. On the other hand, how will he find out that some type is also registered now, but "implicitly" through `AutoRegisteringGraphTypeMapping`?

I don't like that adding this feature can cause additional questions, although it works itself. Need to document such behavior - that `Schema.TypeMappings` does not reflect autoregistration added by new extension methods.

Reply:
All of your points are valid points.  Personally, I would conclude that we should leave `Schema.TypeMappings` and related `RegisterTypeMapping` methods alone, or it will be difficult to write code that builds a schema from code like this:

```cs
var queryType = new ObjectGraphType;
//configure query type

var schema = new Schema();
schema.Query = queryType;
schema.RegisterTypeMapping<MyObject, MyObjectGraphType>();
```

Of course the method could add the mapping to a list of `IGraphTypeMapping` implementations, but like you said, then you can enumerate the added mappings.  Just seems like a small change to leave it as-is, and better for the project overall.

Reply:
On the other hand there is no way to manually add a `IGraphTypeMapping` implementation to a schema without some sort of change here.  So maybe `TypeMappings` returns `IEnumerable<IGraphTypeMapping>` now?

Reply:
The "problem" is a different nature of mappings. `IGraphTypeMapping` is a _factory_ that builds something. You can not say what before you call it. On the other hand `TypeMappings` now is a list of explicitly defined types.

Reply:
Keep as is ?

Reply:
I would keep as it is now, with the changes we added pursuant to this and the below conversation, not as is was when the first comments here were written.